userspace/falco/app/actions/print_syscall_events.cpp

```diff
@@ -55,60 +99,104 @@ static std::vector<event_entry> get_event_entries(bool include_generics, const l
+static bool is_flag_type(ppm_param_type type)
 {
-	if(s.options.list_syscall_events)
+	return (type == PT_FLAGS8 || type == PT_FLAGS16 || type == PT_FLAGS32 ||
+			type == PT_ENUMFLAGS8 || type == PT_ENUMFLAGS16 || type == PT_ENUMFLAGS32);
+}
+
+static void print_param(const struct ppm_param_info *param, bool markdown) {
+	printf("%s **%s**", param_type_to_string(param->type), param->name);
+
+	if (is_flag_type(param->type) && param->info) {
+		auto flag_info = static_cast<const ppm_name_value*>(param->info);
+
+		printf(": ");
+		for (size_t i = 0; flag_info[i].name != NULL; i++) {
+			if (i != 0)
+			{
+				printf(", ");
+			}
+
+			if (markdown) {
+				printf("*%s*", flag_info[i].name);
+			} else {
+				printf("%s", flag_info[i].name);
+			}
+		}
+	}
+}
+
+static void print_events(const std::vector<event_entry> &events, bool markdown)
+{
+	if(markdown)
 	{
-		const auto events = get_event_entries(true, libsinsp::events::all_event_set());
+		printf("Default | Dir | Name | Params \n");
+		printf(":-------|:----|:-----|:-----\n");
+	}
 
-		if(s.options.markdown)
+	for (const auto& e : events)
+	{
+		char dir = e.is_enter ? '>' : '<';
+		if (markdown)
+		{
+			printf(e.available ? "Yes" : "No");
+			printf(" | `%c` | `%s` | ", dir, e.name.c_str());
+		}
+		else
 		{
-			printf("Falco | Dir | Event\n");
-			printf(":-----|:----|:-----\n");
+			printf("%c %s(", dir, e.name.c_str());
 		}
 
-		for (const auto& e : events)
+		for(uint32_t k = 0; k < e.info->nparams; k++)
 		{
-			char dir = e.is_enter ? '>' : '<';
-			if (s.options.markdown)
-			{
-				printf(e.available ? "Yes" : "No");
-				printf(" | %c | **%s**(", dir, e.name.c_str());
-			}
-			else
+			if(k != 0)
 			{
-				printf("%c %s(", dir, e.name.c_str());
+				printf(", ");
 			}
 
-			for(uint32_t k = 0; k < e.info->nparams; k++)
-			{
-				if(k != 0)
-				{
-					printf(", ");
-				}
-
-				printf("%s %s", param_type_to_string(e.info->params[k].type),
-					e.info->params[k].name);
-			}
+			print_param(&e.info->params[k], markdown);
+		}
+		if (markdown)
+		{
+			printf("\n");
+		}
+		else
+		{
 			printf(")\n");
 		}
+	}
+}
+
+falco::app::run_result falco::app::actions::print_syscall_events(falco::app::state& s)
+{
+	if(s.options.list_syscall_events)
+	{
+		const falco::versions_info info(s.offline_inspector);
+		printf("The events below are valid for Falco *Schema Version*: %s\n", info.driver_schema_version.c_str());
+
+		const libsinsp::events::set<ppm_event_code> available = libsinsp::events::all_event_set().diff(sc_set_to_event_set(falco::app::ignored_sc_set()));
+		const struct events_by_category events_bc = get_event_entries_by_category(true, available);
+
+		printf("## Syscall events\n\n");
+		print_events(events_bc.syscalls, s.options.markdown);
+
+		printf("\n\n## Tracepoint events\n\n");
+		print_events(events_bc.tracepoints, s.options.markdown);
+
+		printf("\n\n## Plugin events\n\n");
+		print_events(events_bc.pluginevents, s.options.markdown);
```

Comment:
Do we intend to print all events (regardless of the data source) here? If so, I would change the option name from `--list-syscall-events` to `--list-events`.

cc @falcosecurity/falco-maintainers

Reply:
yes agree with leo i would change it to `--list-events`

Reply:
I noticed that this flag never printed syscall events, but always printed all events. Yes, I introduced it and decided the name 😅 . So I agree with you, let me change it to `--list-events`.